markdown.dtx

```diff
@@ -22317,6 +22313,265 @@ function M.reader.new(writer, options)
+          value["original_count"] = delimeter_length
+      end
+    end
+
+    local current_position = 1
+    local openers_bottom = {
+      ['*'] = { -- character
+        [true] = { -- is opening
+          1, 1, 1 -- lengths
+        },
+        [false] = {
+          1, 1, 1
+        }
+      },
+      ['_'] = {
+        [true] = {
+          1, 1, 1
+        },
+        [false] = {
+          1, 1, 1
+        }
+      }
+    }
+
+    while current_position <= #t do
+      local value = t[current_position]
+      local character = value["character"]
+      local is_opening = value["is_opening"]
+      local is_closing = value["is_closing"]
+      local closing_length_modulo_three = value["original_count"] % 3
+
+      local current_openers_bottom = openers_bottom[character][is_opening][closing_length_modulo_three + 1]
+
+      if is_closing and (value["current_count"] > 0) then
+        local opener_position = find_opener(t, current_openers_bottom, current_position - 2, character, value)
+
+        if (opener_position ~= nil) then
+
+          local opening_delimeter = t[opener_position]
+
+          local current_opening_count = opening_delimeter["current_count"]
+          local current_closing_count = t[current_position]["current_count"]
+
+          if (current_opening_count >= 2) and (current_closing_count >= 2) then
+            opening_delimeter["current_count"] = current_opening_count - 2
+            t[current_position]["current_count"] = current_closing_count - 2
+            fill_strong(t, opener_position, current_position)
+          else
+            opening_delimeter["current_count"] = current_opening_count - 1
+            t[current_position]["current_count"] = current_closing_count - 1
+            fill_emph(t, opener_position, current_position)
+          end
+
+        else
+          openers_bottom[character][is_opening][closing_length_modulo_three + 1] = current_position
+          current_position = current_position + 2
+        end
+      else
+        current_position = current_position + 2
+      end
+    end
+
+    local final_result = collect_emphasis_content(t, 1, #t)
+    return final_result
+  end
+
+  local cont = lpeg.R("\128\191") -- continuation byte
+
+  local function utf8_by_byte_count(n)
+    if (n == 1) then
+      return lpeg.R("\0\127")
+    end
+    if (n == 2) then
+      return lpeg.R("\194\223") * cont
+    end
+    if (n == 3) then
+      return lpeg.R("\224\239") * cont * cont
+    end
+    if (n == 4) then
+      return lpeg.R("\240\244") * cont * cont * cont
+    end
+  end
+
+  local function check_unicode_type(s, i, start_pos, end_pos, chartype)
+    local c = nil
+    local char_length = 0
+    for pos = start_pos, end_pos, 1 do
+      if (start_pos < 0) then
+        char_length = -pos
+      else
+        char_length = pos + 1
+      end
+      c = lpeg.match({ C(utf8_by_byte_count(char_length)) },s,i+pos)
+      if (c ~= nil) and (unicode.utf8.match(c, chartype)) then
```

Comment:
Please, [polyfill the `unicode.utf8.match` method if Selene Unicode is unavailable][1] as discussed [on Discord][2].

[1]: https://github.com/Witiko/markdown/blob/45ee7f11aa8167c0e901d71a77e64165d6db7ca0/markdown.dtx#L1058-L1087
[2]: https://discord.com/channels/@me/1042394066496786473/1080876280045633637